iib/workers/tasks/build_merge_index_image.py

```
@@ -0,0 +1,278 @@
+    """
+    Get a list of to-be-deprecated bundles based on the data from the deprecation list.
+
+    :param list bundles: list of bundles to apply the filter on.
+    :param list deprecation_list: list of deprecated bundle pull specifications (digest).
+    :return: bundles which are to be deprecated.
+    :rtype: list
+    """
+    deprecate_bundles = []
+    for bundle in bundles:
+        if '@sha256:' in bundle['bundlePath']:
+            resolved_bundle = bundle['bundlePath']
+        else:
+            resolved_bundle = _get_resolved_image(bundle['bundlePath'])
+        if resolved_bundle in deprecation_list:
+            deprecate_bundles.append(resolved_bundle)
+
+    log.info(
+        'Bundles that will be deprecated from the index image: %s', ', '.join(deprecate_bundles)
+    )
+    return deprecate_bundles
+
+
+def _deprecate_bundles(
+    bundles, base_dir, binary_image, from_index, overwrite_target_index_token=None,
+):
+    """
+    Deprecate specified bundles from index image. Only Dockerfile is created, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for non-privileged users to use
+        ``overwrite_target_index``. The format of the token must be in the format "user:password".
+    """
+    cmd = [
+        'opm',
+        'index',
+        'deprecatetruncate',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--bundles',
+        ','.join(bundles),
+    ]
+    # TODO catch error of default channel deletion
```

Is this needed?

---

I think we can just let it fail

---

Nope, I just synced up with Shawn and it looks like OPM will interfere if IIB tries to delete a default channel.